src/providers/WorkflowCore.Persistence.EntityFramework/ExtensionMethods.cs

```diff
@@ -27,7 +27,7 @@ internal static PersistedWorkflow ToPersistable(this WorkflowInstance instance,
             persistable.Status = instance.Status;
             persistable.CreateTime = instance.CreateTime;
             persistable.CompleteTime = instance.CompleteTime;
-            persistable.ExecutionPointers = new PersistedExecutionPointerCollection(instance.ExecutionPointers.Count);
+            persistable.ExecutionPointers = persistable.ExecutionPointers ?? new PersistedExecutionPointerCollection(instance.ExecutionPointers.Count);
```

Comment:
How will this behave when new execution pointers are added?
Did you run this against the SQL integration tests?

Reply:
When new execution pointers are added, the method will work as usual, i.e. peristable.ExecutionPointers.FindById(ep.id) at line 32 will return null, therefore a new pointer will be added to the collection. That pointer will be caught by (EF query).AsTrackable() and will be translated into an insert into the database.

From internal tests in a personal project, all workflow instances run/complete as expected on a workflow with multiple parallel branches containing while loops and expecting events. Only difference is a heavily reduced performance impact.

Integration tests/Sql Server unit tests are passing:
![image](https://user-images.githubusercontent.com/14150336/59554746-8f239280-8fb0-11e9-8476-608b9923f980.png)

Reply:
I need to confirm this behavior